lib/spack/spack/stage.py

```diff
@@ -597,6 +597,12 @@ class DIYStage(object):
     """Simple class that allows any directory to be a spack stage."""
 
     def __init__(self, path):
+        if path is None:
+            raise ValueError("Cannot construct DIYStage without a path.")
+        elif not os.path.isdir(path):
+            raise StagePathError("The stage path directory {} does not exist.",
```

Comment:
1. `{}` isn't Python 2.6 compatible (have to use `{0}`, `{1}`, etc. until we drop 2.6)
2. First argument to a Spack Error isn't a format string -- the second argument is "long" text that's printed indented below the short messge.  So this will just print `{}` without substituting, followed by the path indented on the next line.  Just make it say "The stage path directory does not exist:" with a colon and it'll be fine.

Reply:
Oops.  The 2.6 tests passed so I didn't notice.  Will fix.  Thanks.